pkgs/servers/teleport/default.nix

```diff
@@ -61,22 +86,29 @@ buildGoModule rec {
     # https://github.com/NixOS/nixpkgs/issues/132652
     ./test.patch
     ./0001-fix-add-nix-path-to-exec-env.patch
+    ./rdpclient.patch
   ];
 
   # Reduce closure size for client machines
   outputs = [ "out" "client" ];
 
-  preBuild = ''
-    mkdir -p build
-    echo "making webassets"
-    cp -r ${webassets}/* webassets/
-    make lib/web/build/webassets
-
-    ${lib.optionalString withRoleTester
-      "cp -r ${roleTester}/target lib/datalog/roletester/."}
-  '';
+  preBuild =
+    let rustDeps = symlinkJoin {
+      name = "teleport-rust-deps";
+      paths = lib.optional withRdpClient rdpClient
+        ++ lib.optional withRoleTester roleTester;
+    };
+    in
+    ''
+      mkdir -p build
+      echo "making webassets"
+      cp -r ${webassets}/* webassets/
+      make lib/web/build/webassets
+
+      cp -r ${rustDeps}/. .
+    '';
 
-  doCheck = !stdenv.isDarwin;
+  doCheck = false;
```

Comment:
At this point, the test suite contains several tests that do impure things, e.g. it fails looking up the `nixbld` sandbox user for some reason, it tries to spawn the sandbox user's shell (`/noshell`) and can not find it, etc.

Reply:
I wonder if there are lots of tests relying on described features (user lookup and shell) or it might make sense to just disable those impacted tests instead of whole suit? 🤔

Reply:
Sadly, Go provides a way to [include only certain tests](https://stackoverflow.com/questions/48465080/how-do-i-run-specific-golang-test-using-go-test-run) in the run, but not exclude specific tests. We could produce a patch that would remove the problematic tests, or better yet, fix them, but I haven't had the time to properly investigate the failures yet.

Reply:
I was thinking about something like [this](https://gist.github.com/ymatsiuk/3bcb234b3cb7f2f89d46f090aaed30e7#file-disable-failing-tests-patch) 😉 which means we would have to maintain this patch further on 🤔

Reply:
I am not necessarily against it, although we'd have to see about long-term maintenance. Would like to get a second opinion on the best approach here.

Does this also fix tests on darwin (i.e. we can remove `!stdenv.isDarwin`)?

Reply:
Then please add a short comment.